x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/field_types/constant_keyword_type.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { FunctionComponent } from 'react';
+import { i18n } from '@kbn/i18n';
+
+import { documentationService } from '../../../../../../services/documentation';
+import { UseField, Field, JsonEditorField } from '../../../../shared_imports';
+import { getFieldConfig } from '../../../../lib';
+import { NormalizedField } from '../../../../types';
+import { AdvancedParametersSection, EditFieldFormRow, BasicParametersSection } from '../edit_field';
+
+interface Props {
+  field: NormalizedField;
+}
+
+export const ConstantKeywordType: FunctionComponent<Props> = ({ field }) => {
+  return (
+    <>
+      <BasicParametersSection>
+        {/* Value field */}
+        <EditFieldFormRow
+          title={i18n.translate('xpack.idxMgmt.mappingsEditor.constantKeyword.valueFieldTitle', {
+            defaultMessage: 'Set value',
+          })}
+          description={i18n.translate(
+            'xpack.idxMgmt.mappingsEditor.constantKeyword.valueFieldDescription',
+            {
+              defaultMessage:
+                'The value to associate with all documents in the index. If this parameter is not provided, it is set based on the first document that gets indexed.',
+            }
+          )}
+          defaultToggleValue={field.source?.value !== undefined}
+        >
+          <UseField path="value" config={getFieldConfig('value')} component={Field} />
+        </EditFieldFormRow>
+      </BasicParametersSection>
+
+      <AdvancedParametersSection>
+        {/* Meta field */}
+        <EditFieldFormRow
+          title={i18n.translate('xpack.idxMgmt.mappingsEditor.constantKeyword.metaFieldTitle', {
+            defaultMessage: 'Set meta',
```

Comment:
I know this is the name of the parameter, but I'd be inclined to label is as "Set metadata"